gradle-junit-reports/src/main/java/com/palantir/gradle/junit/JunitReportsPlugin.java

```diff
@@ -84,19 +101,14 @@ public void apply(Project project) {
                 .map(RegularFile::getAsFile);
     }
 
-    private static void configureBuildFailureFinalizer(Project rootProject, Provider<Directory> reportsDir) {
-        Provider<RegularFile> targetFileProvider = reportsDir.map(dir -> {
+    private static Provider<File> reportTargetFile(Property<Directory> reportsDir) {
```

Comment:
bit odd to use the lazy interfaces if we eagerly set the value in the parameters